### PR TITLE
8843 blog search func

### LINKED ISSUE #8843: [Blog] - [Search] - Search Functionality
This goal of this ticket is to implement the search functionality of the blogs "search" page.

The search rankings are as follow:

1. Title
2. Category
3. Author
4. Tags
5. (Search/Meta) Description 
7. Body text
8. Date
9. Language

Dev Steps:
--- 

- [ ] Render the search bar template from #8778 into the new search page template from #8772 
- [ ] Implement the search functionality, using the search rankings above.

Acceptance Criteria:
---

- [ ] Users should be able to type a search query in the search bar, when they hit enter or click the icon, the list of search results should filter and update accordingly, displaying the first 6 entries of the list.

### PR DESCRIPTION
**Pull request by tbrlpld**
This PR add the functionality for the blog search. 

When `/blog/search/` is visited without a search query, the page lists the latest 6 blog entries.
When the search bar is used to submit a search query the list of displayed search results is reduced to those that contain the search query. 

Because I was not sure what it means to "search by language" I was not able to implement that functionality. Also, I believe the complexity of "search by date" was not considered when the ticket was estimated. Therefore, I have not implemented that functionality. 

Also, I was not able to create the desired search ranking although the selected search boost values should lead to this ranking. A test is in place to test the desired ranking. But, that test is currently decorated with `@unittest.expectedFailure`. This means that the test suite passes if this test fails. If in the future, for some reason, this test will start passing, the test suite will fail at this test, because it is not failing as expected anymore.  This would be a signal that the desired ranking in the search is working - this should be a good trigger to investigate on what has changed. 

Currently, a maximum of 6 search results is displayed. This will be extended with a "load more" functionality in #8775.

Closes #8843
Related PRs/issues #

**Link to sample test page**: http://localhost:8000/en/blog/search/

## Checklist

_Remove unnecessary checks_

**Tests**
- [x] Is the code I'm adding covered by tests?

**Changes in Models:**
- ~~[ ] Did I update or add new fake data?~~
- ~~[ ] Did I squash my migration?~~
- [x] [Are my changes backward-compatible](https://github.com/mozilla/foundation.mozilla.org/blob/main/docs/workflow.md#django-migrations-what-to-do-when-working-on-backward-incompatible-migrations). If not, did I schedule a deploy with the rest of the team?

**Documentation:**
- [x] Is my code documented? By the tests
- ~~[ ] Did I update the READMEs or wagtail documentation?~~

### CODE REVIEW

**Comment by tbrlpld** [comment]
cc @kevinhowbrook

File: network-api/networkapi/wagtailpages/pagemodels/blog/blog_index.py [hunk 1]
```diff
@@ -239,16 +247,25 @@ def entries_by_topic(self, request, topic, *args, **kwargs):
...
         """Render search results view."""
+
+        query = request.GET.get('q', '')
+
+        context_overrides = {
+            'index_title': 'Search',
+            'entries': self.get_search_entries(query=query)[:6],
+            'query': query,
+        }
+
```
Comment (kevinhowbrook) [inline network-api/networkapi/wagtailpages/pagemodels/blog/blog_index.py:260]: **Suggestion**

Send an extra `no_results = f'no results found' for {query}'` variable through and check `if entries` in the template. At the moment there is no messaging for the user if there are no results

**Review by kevinhowbrook — APPROVED** [review]
:+1:  Looks good to me @tbrlpld just a few minor points, guessing we are at time for this though.

**Review by danielfmiranda — APPROVED** [review]
Hi Tibor! 

Thanks for taking this work on as I know search functionality is quite complex.

+1 to the kudos for adding tests

I also gave the code a look through as well as tested out the search functionality on my local machine and I can confirm that it is working as expected.

Great job! Approved 👍

File: network-api/networkapi/wagtailpages/pagemodels/blog/blog_index.py (hunk 1 above)
Comment (tbrlpld) [reply network-api/networkapi/wagtailpages/pagemodels/blog/blog_index.py:260]: Good catch. There is #8776 to add a template for this, but I could definitely handle the case already and just fill in the template 👍

File: network-api/networkapi/wagtailpages/tests/test_blog.py [hunk 2]
```diff
@@ -1,60 +1,292 @@
...
+        cls.search_term = "Aweirdsearchquery"
+
     def test_route_success(self):
-        blog_index = blog_factories.BlogIndexPageFactory(parent=self.homepage)
-        url = blog_index.get_url() + blog_index.reverse_subpage("search")
+        url = self.blog_index.get_url() + self.blog_index.reverse_subpage("search")
 
         response = self.client.get(path=url)
 
         self.assertEqual(response.status_code, HTTPStatus.OK)
```
[the remark this answers is not in the document]
Comment (tbrlpld) [reply network-api/networkapi/wagtailpages/tests/test_blog.py:36]: Good call. I'll add that.

**Comment by tbrlpld** [comment]
@danielfmiranda @kevinhowbrook I was able to update the search ranking today because I learned that the Postgres search (that we are using) only allows 4 total levels for the boost values. I only had to adjust the boost values and update the ranking tests (that previously had been expected failures). Maybe you want to take another look.

**Comment by tbrlpld** [comment]
I am gonna go ahead with it the way it is because the changes are minimal compared to the approved version and my next tickets depend on it. Let me know if this needs any followup.

**Pull request merged**